front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/checkCurrentConfig.ts

```diff
@@ -145,22 +146,28 @@ const checkCurrentConfig = (
     rollingStockComfort,
     initialSpeed: initialSpeed ? kmhToMs(initialSpeed) : 0,
     usingElectricalProfiles,
-    path: compact(pathSteps).map((step) => {
-      // TODO use lodash pick
-      const {
-        arrival,
-        locked,
-        stopFor,
-        positionOnPath,
-        coordinates,
-        name,
-        ch,
-        metadata,
-        kp,
-        receptionSignal,
-        theoreticalMargin,
-        ...stepLocation
-      } = step;
+    path: compact(pathSteps).map((step: PathStep) => {
+      const { ch } = step;
+      const stepLocation = omit(
```

Comment:
Shouldn't this be a `pick()` with `track` + `offset` + `trigram` + all other fields which can be in `PathItemLocation`, rather than `omit()`?

Reply:
Well, both work. What is your argument in favor of `pick` rather than `omit`?

I chose to use `omit` because the common factor between `PathStep` and `TrainScheduleBase['path']` is `PathItemLocation`, which is a union type, so not something easy to construct properly using `pick`:

```typescript
export type PathItemLocation =
  | TrackOffset
  | {
      operational_point: string;
    }
  | {
      /** An optional secondary code to identify a more specific location */
      secondary_code?: string | null;
      trigram: string;
    }
  | {
      /** An optional secondary code to identify a more specific location */
      secondary_code?: string | null;
      /** The [UIC](https://en.wikipedia.org/wiki/List_of_UIC_country_codes) code of an operational point */
      uic: number;
    };
```

Reply:
I think the old code was better than `omit()`, because in both cases we're listing all of the fields we don't want, and when using destructuring there's no need to cast. (As a bonus, it's vanilla JS.)

I'd prefer to use `pick()` because there are more fields we don't want than fields we want to keep, and we're more likely to add fields in `PathStep` then we are to add some in `PathItemLocation`.

> a union type, so not something easy to construct properly using pick

A union type should be easy to construct from `pick()` no? Basically just:

```ts
pick(pathStep, 'track', 'offset', 'secondary_code', 'trigram', 'uic')
```

Any field not present in the original value still won't be present in the final value.

Reply:
I also think `pick` seems simpler bu not sure if that's why @jacomyal prefers `omit`, but as when using `pick` we don't have typescript auto completion for the `PathItemLocation` properties, it might result to some weird things like `pick(pathStep, 'track', 'offffset', 'secondaryCode', 'trigram', 'uic', 'blob')` without getting any error as we have to use a cast ?

Reply:
To me `omit()` is the worst of both worlds: no upside over object destructuring and same downsides as `pick()`.

(Ideally I'd prefer to isolate the `PathItemLocation` in a prop of `PathStep`...)